File: javascript/app/libraries/vector.js
function Vector() {};

Vector.module = function(vector) {
    var temp = 0,
        length = vector.length;
    for(var i = 0; i < length; i++) {
        temp += vector[i] * vector[i];
    }
    return Math.sqrt(temp);
};

Vector.multiplication = function(vector_a, vector_b) {
    var temp = 0,
        length = vector_a.length;
    for(var i = 0; i < length; i++) {
        temp += vector_a[i] * vector_b[i];
    }
    return temp;
};

Vector.normalize = function(vector) {
    var temp = [],
        length = vector.length,
        module = this.module(vector);
    for(var i = 0; i < length; i++) {
        temp[i] = vector[i]/module;
    }
    return temp;
};

Vector.angle = function(vector_a, vector_b) {
    return this.multiplication(vector_a, vector_b)/(this.module(vector_a) * this.module(vector_b));
};

File: python/classes/libraries/Vector.py
from math import sqrt
from functools import reduce

class Vector:
    @staticmethod
    def module(vector):
        return sqrt(reduce((lambda x, y: x+y), map(lambda x: x**2, vector)))
    @staticmethod
    def multiplication(vector_a, vector_b):
        return reduce((lambda x, y: x+y), map(lambda x, y: x*y, vector_a, vector_b))
    @staticmethod
    def angle(vector_a, vector_b):
        return Vector.multiplication(vector_a, vector_b)/(Vector.module(vector_a) * Vector.module(vector_b))
    @staticmethod
    def normalize(vector):
        length = Vector.module(vector)
        return list(map(lambda x: x/length, vector))